fence/blueprints/google.py

```diff
@@ -769,6 +770,18 @@ def _get_google_project_id_error_status(validity_info):
 
     user_has_access = validity_info.get("user_has_access")
     if not user_has_access:
+        if not validity_info.google_project_id or validity_info.google_project_id == "":
```

Comment:
```suggestion
        if not validity_info.google_project_id:
```
empty string evaluates to False-y

Reply:
I feel like it might be good to refactor this whole function to look more like the others where it inits a successful response at the top and then replaces and returns early if error, lemme know what you think
```
def _get_google_project_id_error_status(validity_info):
    has_access = validity_info.get("monitor_has_access")
    user_has_access = validity_info.get("user_has_access")
    valid_parent_org = validity_info.get("valid_parent_org")
    valid_member_types = validity_info.get("valid_member_types")
    members_exist_in_fence = validity_info.get("members_exist_in_fence")
    service_accounts_validity = validity_info.get("service_accounts")

    response = {
        "status": 200,
        "error": None,
        "error_description": "",
        "membership_validity": {
            "valid_member_types": valid_member_types,
            "members_exist_in_fence": members_exist_in_fence,
        },
        "service_account_validity": {},
    }

    if not has_access:
        response["status"] = 404
        response["error"] = ValidationErrors.MONITOR_NOT_FOUND
        response["error_description"] = (
            "Fence's monitoring service account "
            "does not have access to the project and/or the necessary "
            "Google APIs are not enabled."
        )
        return response

    if not user_has_access:
        response["status"] = 403
        response["error"] = ValidationErrors.UNAUTHORIZED_USER
        response["error_description"] = (
            "Current user is not an authorized member on the provided "
            "Google Project."
        )
        return response

    for sa_account_id, sa_validity in service_accounts_validity:
        if sa_account_id != validity_info.new_service_account:
            response["service_account_validity"][sa_account_id] = sa_validity.get_info()
            if not sa_validity:
                response["status"] = 403
                response["error"] = ValidationErrors.UNAUTHORIZED
                response[
                    "error_description"
                ] = "Project has one or more invalid service accounts. "

    if not valid_parent_org:
        response["status"] = 403
        response["error"] = ValidationErrors.UNAUTHORIZED
        response["error_description"] += "Project has parent organization. "

    if not valid_member_types:
        response["status"] = 403
        response["error"] = ValidationErrors.UNAUTHORIZED
        response["error_description"] += "Project has invalid member types. "

    if not members_exist_in_fence:
        response["status"] = 403
        response["error"] = ValidationErrors.UNAUTHORIZED
        response[
            "error_description"
        ] += "Not all Google project members exist in fence."

    return response
```